### PR TITLE
change sdk to 0.9.2

### PR DESCRIPTION
**Pull request by rudewalt**

### CODE REVIEW

File: pool/src/test/pool_test.rs [hunk 1]
```diff
@@ -247,13 +262,17 @@ fn init_reserve_second_time() {
...
-            .try_init_reserve(&sut.token().address, &init_reserve_input)
-            .unwrap_err()
-            .unwrap(),
-        Error::ReserveAlreadyInitialized
-    )
+    //TODO: check error after soroban fix
+    sut.pool
+        .init_reserve(&sut.token().address, &init_reserve_input);
+
+    // assert_eq!(
```
Comment (artur-abliazimov) [inline pool/src/test/pool_test.rs:269]: Should this be removed?

File: s-token/src/lib.rs [hunk 2]
```diff
@@ -433,10 +407,16 @@ impl SToken {
...
-        if allowance < amount {
+        if allowance.amount < amount {
             panic!("s-token: insufficient allowance");
         }
-        write_allowance(e, from, spender, allowance - amount);
+        write_allowance(
+            e,
+            from,
+            spender,
+            allowance.amount - amount,
```
Comment (artur-abliazimov) [inline s-token/src/lib.rs:417]: Do we need to do the `checked_sub` as we usually do?

File: s-token/src/lib.rs [hunk 3]
```diff
@@ -85,52 +88,23 @@ impl STokenTrait for SToken {
...
+    /// - expiration_ledger - The time when allowance will be expired.
     ///
     /// # Panics
     ///
     /// Panics if the caller is not authorized.
     /// Panics if the amount is negative.
     /// Panics if the updated allowance exceeds the maximum value of i128.
     ///
-    fn increase_allowance(e: Env, from: Address, spender: Address, amount: i128) {
+    fn approve(e: Env, from: Address, spender: Address, amount: i128, expiration_ledger: u32) {
```
Comment (artur-abliazimov) [inline s-token/src/lib.rs:99]: Do I understand correctly that the `approve` is now responsible for allowance?
I thought that the `approve` usually have different meaning - to approve some other party to transfer tokens.

File: s-token/src/lib.rs [hunk 4]
```diff
@@ -85,52 +88,23 @@ impl STokenTrait for SToken {
...
         from.require_auth();
 
         require_nonnegative_amount(amount);
 
-        let allowance = read_allowance(&e, from.clone(), spender.clone());
-        let new_allowance = allowance
-            .checked_add(amount)
-            .expect("Updated allowance doesn't fit in an i128");
+        e.storage().instance().bump(INSTANCE_BUMP_AMOUNT);
```
Comment (artur-abliazimov) [inline s-token/src/lib.rs:104]: Let's move all the storage interactions to `storage.rs`.

Comment (artur-abliazimov) [inline s-token/src/lib.rs:104]: Should the bump be consistent to the `expiration_ledger`?

File: s-token/src/lib.rs [hunk 5]
```diff
@@ -75,7 +78,7 @@ impl STokenTrait for SToken {
     /// The amount of tokens that the `spender` is allowed to withdraw from the `from` address.
     ///
     fn allowance(e: Env, from: Address, spender: Address) -> i128 {
-        read_allowance(&e, from, spender)
+        read_allowance(&e, from, spender).amount
     }
 
     /// Increases the allowance for a spender to withdraw from the `from` address by a specified amount of tokens.
```
Comment (artur-abliazimov) [inline s-token/src/lib.rs:84]: Let's align the description with the new parameter.

File: s-token/src/lib.rs [hunk 6]
```diff
@@ -9,9 +9,12 @@ use crate::storage::*;
 use common_token::{balance::*, require_nonnegative_amount, storage::*, verify_caller_is_pool};
 use pool_interface::LendingPoolClient;
 use s_token_interface::STokenTrait;
-use soroban_sdk::{contractimpl, token, Address, Bytes, Env};
+use soroban_sdk::{contract, contractimpl, token, Address, Env, String};
 use soroban_token_sdk::TokenMetadata;
 
+pub(crate) const INSTANCE_BUMP_AMOUNT: u32 = 34560; // 2 days
```
Comment (artur-abliazimov) [inline s-token/src/lib.rs:15]: Must this be in `common`?

File: mocks/price-feed-mock/src/storage.rs [hunk 7]
```diff
@@ -9,15 +9,15 @@ pub enum DataKey {
 pub fn write_asset_price(env: &Env, asset: Address, price: i128) {
     let data_key = DataKey::Price(asset);
 
-    env.storage().set(&data_key, &price);
+    env.storage().persistent().set(&data_key, &price);
```
Comment (artur-abliazimov) [inline mocks/price-feed-mock/src/storage.rs:12]: Should we use `instance` instead?

File: pool/src/storage.rs [hunk 8]
```diff
@@ -16,102 +16,104 @@ pub enum DataKey {
...
-        .unwrap()
+        .ok_or(Error::NoPriceFeed)
 }
 
 pub fn write_price_feed(env: &Env, feed: Address, assets: &Vec<Address>) {
     for asset in assets.iter() {
-        let data_key = DataKey::PriceFeed(asset.unwrap());
-        env.storage().set(&data_key, &feed);
+        let data_key = DataKey::PriceFeed(asset);
+        env.storage().persistent().set(&data_key, &feed);
```
Comment (artur-abliazimov) [inline pool/src/storage.rs:98]: Should we use `instance` instead?

**Review by mn13 — APPROVED** [review]

File: pool/src/test/pool_test.rs (hunk 1 above)
Comment (rudewalt) [reply pool/src/test/pool_test.rs:269]: No, I left it here to uncomment when issue will be fixed

File: s-token/src/lib.rs (hunk 6 above)
Comment (rudewalt) [reply s-token/src/lib.rs:15]: No, only s-token used the const

**Pull request merged**